src/heretic/g_game.c

```diff
@@ -1775,7 +1782,28 @@ void G_RecordDemo(skill_t skill, int numplayers, int episode, int map,
     *demo_p++ = episode;
     *demo_p++ = map;
 
-    for (i = 0; i < MAXPLAYERS; i++)
+    // Write special parameter bits onto player one byte.
+    // This aligns with vvHeretic demo usage:
+    //   0x20 = -respawn
+    //   0x10 = -longtics
+    //   0x02 = -nomonsters
+
+    *demo_p = 1; // assume player one exists
+    if (respawnparm)
+    {
+        *demo_p += 32;
+    }
+    if (longtics)
```

Comment:
Vanilla Heretic and Hexen do not store any information when it comes to special parameters: demo headers consist only of skill/episode/map and player counts (and classes in Hexen). vvHeretic probably wanted to account not just for new parameters like `-longtics` but existing ones as well. We're quite lucky they chose to store single-bit information in an entire byte so there was a place to stuff everything else.

I tested some demos in vanilla, they seem to work as expected: sync is fine as long as you include the extra parameter in the command line on playback.